pkg/controller/iam/auditconfig/iamauditconfig_controller_integration_test.go

```diff
@@ -606,6 +608,15 @@ func newIAMAuditConfigFixture(t *testing.T, refResource *unstructured.Unstructur
 	}
 }
 
+func reconcileIAMAuditConfig(ctx context.Context, t *testing.T, reconciler *testreconciler.TestReconciler, policy *iamv1beta1.IAMAuditConfig, expectedResult reconcile.Result, expectedErrorRegex *regexp.Regexp) {
```

Comment:
The change looks good to me, though have you verified it locally?

Reply:
I have not! What is the correct way to run these tests locally? Ideally it would be possible to just run a single test case locally (instead of the whole suite).

Reply:
Ok I figured out how to run the IAM tests locally using vscode launch configurations like the following:

```
{
	"name": "dynamic - basic-spannerinstance",
	"type": "go",
	"request": "launch",
	"mode": "test",
	"program": "/usr/local/google/home/jasonvigil/code/k8s-config-connector/pkg/controller/dynamic/",
	"buildFlags": [
		"-tags=integration",
	],
	"args": [
		"-test.v",
		"-test.timeout", "1600s",
		"-test.count=1",
		"-test.run", "TestCreateNoChangeUpdateDelete/spanner/^basic-spannerinstance$"
	],
},
```

And, ended up catching a bug!

The testreconciler could not recognize the IAM resources because they had empty GVKs. The GVKs were erased by previously calling `kubeClient.Create` with the typed resources, which uses the typed client to decode the result back into the object and erases the GVK [[ref](https://github.com/kubernetes/apimachinery/blob/c463db196543ee48d2353c6ca3325de6454ad075/pkg/runtime/helper.go#L254-L263)].

The dynamic controller tests are immune to this issue because they use unstructured resources instead of typed resources. Therefore, they use the unstructured kube client, which decodes the result directly back into the provided unstructured object, and does not erase the GVK [[ref](https://github.com/kubernetes/apimachinery/blob/c463db196543ee48d2353c6ca3325de6454ad075/pkg/runtime/serializer/json/json.go#L171-L180)].

I was able to verify this fix for policy, partialpolicy, and policymember. However, I am struggling to run the auditconfig tests locally. Is there some special configuration necessary? Here is the error I am getting (on both master and this feature branch):

```
=== NAME  TestReconcileIAMAuditConfigResourceLevelCreate/resourcemanager/basic-folderinfolder
    runner.go:74: reconcile for Folder:jw5yhppa3xsx4yi/folder-jw5yhppa3xsx4yi took 110.626384ms, result was ({false 0s}, Update call failed: error expanding resource configuration for kind Folder: error getting referenced resource from API server: resource reference is missing required 'name' field)
```

Can you please help me to run this test locally?

Reply:
AFAIK, we don't have any filter to just run a single test case in these integration tests. Feel free to hack them locally for testing purpose if you like, or support them as a new feature. Or you can let the whole suite run in the background. Note that because not all resources support specific IAM features, the test suite _may_ not be that large.

Here is an example command:
```
go test -v -tags=integration ./pkg/controller/iam/auditconfig -test.run ^TestReconcileIAMAuditConfigResourceLevelCreate$ -timeout 900s
```

Reply:
Ah, glad to see you figured out how to test it.

Regarding `resource reference is missing required 'name' field` error message, can you verify that you've configured the `TEST_FOLDER_ID` env var? I think it looks for `folderRef.name` because `folderRef.external` is empty.

Reply:
Ah, thank you @maqiuyujoyce ! Was missing th env var. I have now verified this change with one test case from each of the IAM categories.